langchain-ai/langgraph
# DOC: Fix issue in manage-conversation-history.ipynb where unbound model is used for LLM invocations

See a link to the published example [here](https://langchain-ai.github.io/langgraph/how-tos/memory/manage-conversation-history/#setup).

The example walks through setting up LangGraph with memory. As part of the standard setup, a model is created using `ChatAnthropic`. Tools are bound to the model and saved in a varliable named `bound_model`. The original model, stored with a variable named `model` (_without_ tool access), is then used as part of the `call_model` function.

Although this does not affect the outcome of the example as written, it is still important to use the correct model as many individuals will pattern after this demo when adding memory in their own applications.

### Reproduction Steps

1. Run the notebook as written, except when providing human input, enter something like "What tools do you have available to you?"
2. Observe the output, which will likely include hallucinated tool names, and _not_ the `search` tool as exemplified in the notebook.

#### Screenshot Example
![2024-07-21_20-50](https://github.com/user-attachments/assets/5fc85cb1-c1ee-4e57-baef-9bc128466bdd)


### Testing Steps

1. Pull the code with the commit added as part of this pull request.
2. Modify the human input to enter something like "What tools do you have available to you?"
4. Observe the output, which should now include the `search` tool along with the `multi_tool_use` tool

#### Screenshot Example
![2024-07-21_20-50_1](https://github.com/user-attachments/assets/53dc3569-99bb-49bb-9ae7-285b86f90828)
